Find a functionally equivalent code:android.animation.ValueAnimator.start:COMMENT
Method Modifier: private     
Comment:/**
 * Start the animation playing. This version of start() takes a boolean flag that indicates
 * whether the animation should play in reverse. The flag is usually false, but may be set
 * to true if called from the reverse() method.
 *
 * <p>The animation started by calling this method will be run on the thread that called
 * this method. This thread should have a Looper on it (a runtime exception will be thrown if
 * this is not the case). Also, if the animation will animate
 * properties of objects in the view hierarchy, then the calling thread should be the UI
 * thread for that view hierarchy.</p>
 *
 * @param playBackwards Whether the ValueAnimator should start playing in reverse.
 */

@@ -8,13 +8,15 @@
     mStarted = true;
     mStartedDelay = false;
     mPaused = false;
+    // in case the scale factor has changed since creation time
+    updateScaledDuration();
     AnimationHandler animationHandler = getOrCreateAnimationHandler();
     animationHandler.mPendingAnimations.add(this);
     if (mStartDelay == 0) {

